process_data.sh

```diff
@@ -259,107 +285,107 @@ for contrast in "${contrasts[@]}"; do
-    # Registration
-    # ------------------------------------------------------------------------------
-    sct_register_multimodal -i ${file_path}.nii.gz -d ./anat/${file_t2}.nii.gz -iseg ${fileseg}_pad.nii.gz -dseg ./anat/${file_t2_seg}.nii.gz -param step=1,type=seg,algo=slicereg,metric=MeanSquares,iter=10,poly=2 -qc ${PATH_QC} -qc-subject ${SUBJECT} -o ${file_path}_reg.nii.gz
-    warping_field=${type}warp_${file}2${file_t2}
-
-    # Generate SC segmentation coverage and register to T2w
-    # ------------------------------------------------------------------------------
-    sct_create_mask -i ${file_path}.nii.gz -o ${file_path}_ones.nii.gz -size 500 -p centerline,${fileseg}.nii.gz
-    # Bring coverage mask to T2w space
-    sct_apply_transfo -i ${file_path}_ones.nii.gz -d ./anat/${file_t2}.nii.gz -w ${warping_field}.nii.gz -x linear -o ${file_path}_ones_reg.nii.gz
-    # Bring SC segmentation to T2w space
-    sct_apply_transfo -i ${fileseg}.nii.gz -d ./anat/${file_t2_seg}.nii.gz -w ${warping_field}.nii.gz -x linear -o ${fileseg}_reg.nii.gz
-  done
-  # Create coverage mask for T2w
-  sct_create_mask -i ./anat/${file_t2}.nii.gz -o ./anat/${file_t2}_ones.nii.gz -size 500 -p centerline,./anat/${file_t2_seg}.nii.gz
-  # Create soft SC segmentation
-  # ------------------------------------------------------------------------------
-  # Sum all coverage images
-  contrasts_coverage=${inc_contrasts[@]/%/"_ones_reg.nii.gz"}
-  sct_maths -i ./anat/${file_t2}_ones.nii.gz -add $(eval echo ${contrasts_coverage[@]}) -o sum_coverage.nii.gz
-  # Sum all segmentations
-  contrasts_seg=${inc_contrasts[@]/%/"_seg_reg.nii.gz"}
-  sct_maths -i ./anat/${file_t2_seg}.nii.gz -add $(eval echo ${contrasts_seg[@]}) -o sum_sc_seg.nii.gz
-  # Divide sum_sc_seg by sum_coverage
-  sct_maths -i sum_sc_seg.nii.gz -div sum_coverage.nii.gz -o ./anat/${file_t2}_seg_soft.nii.gz
+# Generate softsegs
+# Create mask for regsitration
+find_manual_seg ${file_t2} 'anat' 't2'
+file_t2_seg="${file_t2}_seg"
+file_t2_mask="${file_t2_seg}_mask"
+sct_create_mask -i ./anat/${file_t2}.nii.gz -p centerline,./anat/${file_t2_seg}.nii.gz -size 55mm -o ./anat/${file_t2_mask}.nii.gz 
+
+# Label intervertebral discs of T2w
+label_if_does_not_exist ./anat/${file_t2} ./anat/${file_t2_seg}
+file_t2_seg_labeled="${file_t2_seg}_labeled"
+file_t2_discs="${file_t2_seg}_labeled_discs"
+
+# Generate QC report to assess vertebral labeling
+sct_qc -i ./anat/${file_t2}.nii.gz -s ./anat/${file_t2_seg_labeled}.nii.gz -p sct_label_vertebrae -qc ${PATH_QC} -qc-subject ${SUBJECT}
 
-  file_softseg=./anat/"${file_t2}_seg_soft"
+# Loop through available contrasts
+for file_path in "${inc_contrasts[@]}";do
+  # Find contrast to do segmentation
+  if [[ $file_path == *"T1w"* ]];then
+      contrast_seg="t1"
+  elif [[ $file_path == *"T2star"* ]];then
+      contrast_seg="t2s"
+  elif [[ $file_path == *"T1w_MTS"* ]];then
+      contrast_seg="t1"
+  elif [[ $file_path == *"MTon_MTS"* ]];then
```

Comment:
you can also add the MToff

Reply:
As discussed, the segmentation with sct_deepseg_sc of MToff was to bad and would have required a lot of manual segmentation.